### ISSUE TITLE
Allow Filtering IP Addresses by Related Virtual Machine Role

### ISSUE THREAD

**Issue by robduffy2010** [netbox-community/netbox#18058]
### NetBox version

v4.1.6

### Feature type

New functionality

### Triage priority

I volunteer to perform this work (if approved)

### Proposed functionality

Right, now it's possible to filter an IP address by it's related virtual machine name or ID.  We would like to be able to also filter by the related virtual machine role. 

https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/filtersets.py#L613

### Use case

It can be very convenient to find all IPs connected to VMs with a particular role.  For example, all Radius, AD or DNS server IP addresses.  

### Database changes

NA

### External dependencies

NA

**Comment by robduffy2010** [comment]
@bctiemann could you take a look at the pull request for this one when you have some time?  I originally forgot to link the pull request to an issue so it might have been missed.

**Comment by bctiemann** [comment]
@robduffy2010 thanks for your work on this. There's been some internal discussion around how supportable this kind of change is, where this would set a precedent for arbitrary fields on related models to be broadly thought of as filterable through the addition of simple, incremental, even in-spirit additions like this. I think that where we've landed is that having `pk` and `name` is the limit of what fields on related objects we want to be filterable per the pervasive systemwide pattern, and beyond that we shouldn't add the functionality unless/until we develop a way to make it sustainable at scale (i.e. some kind of auto-config framework that eliminates the need for maintenance).

What we want to avoid in particular is inviting future contributors to see opportunities to add arbitrary filter fields like this one, which would certainly happen (considering how comparatively little effort it takes to extend the classes piecemeal). While convenient, it's not maintainable. My apologies!

**Comment by robduffy2010** [comment]
While this is a good use case, I understand the reasoning and not wanting to set a precedent.  I'll go ahead and close the pull request. 

As a workaround, I'd like to search for VM names which include a specific string.  However, this doesn't seem to work according to the documentation.  Is this as intended or is it a bug?

**Comment by robduffy2010** [comment]
Just to clarify, I'm referring to the __ic filter referenced in the documentation.

https://demo.netbox.dev/static/docs/rest-api/filtering/#string-fields

The query I'm using is. 

https://netbox.company.tld/api/ipam/ip-addresses/?virtual_machine__ic=rad

**Comment by jeremystretch** [comment]
Hi @robduffy2010, are you still working on this?

**Comment by robduffy2010** [comment]
@jeremystretch the change was initially approved so I sent in a pull request. It was then rejected as the core team changed their mind regarding which types of changes ought to be permitted. If it's now going to be accepted, I can submit a new pull request.

**Comment by jeremystretch** [comment]
Yeah I can definitely see how this sort of thing could spiral out of control without a clear, scalable pattern in place. Looks like something for future consideration.
